src/test/java/net/obvj/junit/utils/matchers/ExceptionMatcherTest.java

```diff
@@ -130,6 +134,36 @@ public void withCause_nullButHasCause_fails()
                 throwsException(IllegalArgumentException.class).withCause(null));
     }
 
+    @Test
+    public void withMessage_equalToAndStringMatching_suceeds()
+    {
+        assertThat(() -> RUNNABLE_THROWS_ISE_WITH_MESSAGE_AND_NO_CAUSE.run(),
+                throwsException(IllegalStateException.class).withMessage(equalTo(MESSAGE_ERR_0001_FULL)));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void withMessage_equalToAndStringNotMatching_fails()
+    {
+        assertThat(() -> RUNNABLE_THROWS_ISE_WITH_MESSAGE_AND_NO_CAUSE.run(),
+                throwsException(IllegalStateException.class).withMessage(equalTo(MESSAGE1)));
+    }
+
+    @Test
+    public void withMessage_combinedMatcherAndStringMatching_suceeds()
+    {
+        assertThat(() -> RUNNABLE_THROWS_ISE_WITH_MESSAGE_AND_NO_CAUSE.run(),
+                throwsException(IllegalStateException.class)
+                        .withMessage(either(startsWith(ERR_0001)).or(containsAny(MESSAGE1).ignoreCase())));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void withMessage_combinedMatcherAndStringNotMatching_fails()
+    {
+        assertThat(() -> RUNNABLE_THROWS_ISE_WITH_MESSAGE_AND_NO_CAUSE.run(),
+                throwsException(IllegalStateException.class)
+                        .withMessage(either(startsWith(MESSAGE1)).or(containsAny(MESSAGE2).ignoreCase())));
+    }
+
```

Comment:
Add more tests to cover the scenario of passing a null matcher reference to the ExceptionMatcher (both for an exception with an actual message and an exception with no message